File: backend/app/schemas/bot.py
"""
Bot 机器人 Schema

Pydantic 模型定义
"""

from typing import Literal, Optional

from pydantic import BaseModel, Field


class OneBot11ConfigSchema(BaseModel):
    """OneBot v11 渠道配置"""
    listen_host: str = Field(default="0.0.0.0", description="WebSocket 服务端监听地址")
    listen_port: int = Field(default=6199, description="WebSocket 服务端监听端口")
    token: Optional[str] = Field(default=None, description="鉴权 Token")
    message_post_format: Literal["array", "string"] = Field(default="array", description="消息上报格式")
    self_message: bool = Field(default=False, description="是否接收自身消息")
    extra_config: Optional[dict] = Field(default=None, description="额外配置")


class BotCreate(BaseModel):
    """创建机器人请求"""
    bot_id: str = Field(..., description="机器人唯一标识")
    bot_name: str = Field(..., description="机器人名称")
    channel_type: Literal["onebot11", "feishu", "dingtalk", "telegram", "qqofficial"] = Field(
        ..., description="渠道类型"
    )
    preset_id: str = Field(..., description="关联的预设方案 ID")
    is_active: bool = Field(default=True, description="是否启用")
    channel_config: Optional[dict] = Field(default=None, description="渠道特定配置（供将来扩展）")


class BotUpdate(BaseModel):
    """更新机器人请求"""
    bot_name: Optional[str] = Field(default=None, description="机器人名称")
    preset_id: Optional[str] = Field(default=None, description="关联的预设方案 ID")
    is_active: Optional[bool] = Field(default=None, description="是否启用")


class OneBot11ConfigCreate(BaseModel):
    """创建 OneBot v11 配置"""
    listen_host: str = Field(default="0.0.0.0", description="WebSocket 服务端监听地址")
    listen_port: int = Field(default=6199, description="WebSocket 服务端监听端口")
    token: Optional[str] = Field(default=None, description="鉴权 Token")
    message_post_format: Literal["array", "string"] = Field(default="array", description="消息上报格式")
    self_message: bool = Field(default=False, description="是否接收自身消息")
    extra_config: Optional[dict] = Field(default=None, description="额外配置")


class OneBot11ConfigUpdate(BaseModel):
    """更新 OneBot v11 配置"""
    listen_host: Optional[str] = Field(default=None, description="WebSocket 服务端监听地址")
    listen_port: Optional[int] = Field(default=None, description="WebSocket 服务端监听端口")
    token: Optional[str] = Field(default=None, description="鉴权 Token")
    message_post_format: Optional[Literal["array", "string"]] = Field(default=None, description="消息上报格式")
    self_message: Optional[bool] = Field(default=None, description="是否接收自身消息")
    extra_config: Optional[dict] = Field(default=None, description="额外配置")
    is_active: Optional[bool] = Field(default=None, description="是否启用")


class BotResponse(BaseModel):
    """机器人响应"""
    bot_id: str
    bot_name: str
    channel_type: str
    preset_id: str
    is_active: bool

    class Config:
        from_attributes = True


class BotDetailResponse(BotResponse):
    """机器人详情响应（包含渠道配置）"""
    channel_config: Optional[dict] = None

    class Config:
        from_attributes = True

File: frontend/src/api/bot.ts
import { http } from "./request";

export type ChannelType = "onebot11" | "feishu" | "dingtalk" | "telegram" | "qqofficial";

export interface BotResponse {
  bot_id: string;
  bot_name: string;
  channel_type: ChannelType;
  preset_id: string;
  is_active: boolean;
}

export interface BotDetailResponse extends BotResponse {
  channel_config?: {
    listen_host?: string;
    listen_port?: number;
    token?: string;
    message_post_format?: "array" | "string";
    self_message?: boolean;
    is_active?: boolean;
  };
}

export interface BotCreate {
  bot_id: string;
  bot_name: string;
  channel_type: ChannelType;
  preset_id: string;
  is_active?: boolean;
}

export interface BotUpdate {
  bot_name?: string;
  preset_id?: string;
  is_active?: boolean;
}

export const botApi = {
  list: () => http.get<{ data: BotResponse[] }>("/bots"),

  get: (bot_id: string) => http.get<{ data: BotDetailResponse }>(`/bots/${bot_id}`),

  create: (data: BotCreate) => http.post<{ data: BotResponse }>("/bots", data),

  update: (bot_id: string, data: BotUpdate) =>
    http.put<{ data: BotResponse }>(`/bots/${bot_id}`, data),

  delete: (bot_id: string) => http.delete(`/bots/${bot_id}`),
};

export const channelTypeLabels: Record<ChannelType, string> = {
  onebot11: "OneBot v11",
  feishu: "飞书",
  dingtalk: "钉钉",
  telegram: "Telegram",
  qqofficial: "QQ 官方",
};

export const channelTypeColors: Record<ChannelType, string> = {
  onebot11: "#409eff",
  feishu: "#67c23a",
  dingtalk: "#e6a23c",
  telegram: "#909399",
  qqofficial: "#f56c6c",
};